projects/bootloader/src/main.c

```diff
@@ -8,6 +8,7 @@
 #include "flash.h"
 #include "interrupt.h"
 #include "jump_to_application.h"
+#include "jump_to_bootloader.h"
```

Comment:
So the next steps for this module is that we want to jump from the bootloader to the application when we receive a datagram message while in an application. This will be in can_fsm.c, in the `if (rx_msg.source_id == SYSTEM_CAN_DEVICE_BOOTLOADER)` if statement. Essentially, when that if statement is reached:
* if we're in an application running on the bootloader, in which case we should jump to the bootloader so that the bootloader can receive subsequent datagram messages
* oherwise, we should call `bootloader_can_receive` there like we do now so that the bootloader can process the message (if necessary)

So there are a couple of parts to doing this:
1. We'll need to move `jump_to_bootloader` to a library, perhaps called `ms-bootloader`, so can_fsm.c can reach it. Perhaps move bootloader_mcu.h to that library too.
2. We'll need a way to determine which of the three scenarios above we're in. This is a bit tricky: we'll need a preprocessor symbol that's defined when we're compiling for an application running on the bootloader. You can do this by adding a `-D<symbolname>` argument to the C compiler flags when compiling for the application, which currently you can do in platform/stm32f0xx/platform.mk by adding `CFLAGS += -D<symbolname>` to the `ifeq ($(MAKECMDGOALS),temp-bootloader-write)` if statement (replace `<symbolname> with the symbol you want defined when on the application only). Then you can test with `#ifdef` in can_fsm.c.